src/main/java/stashpullrequestbuilder/stashpullrequestbuilder/StashBuilds.java

```diff
@@ -49,15 +50,22 @@ public void onCompleted(AbstractBuild build, TaskListener listener) {
             return;
         }
         Result result = build.getResult();
-        JenkinsLocationConfiguration globalConfig = new JenkinsLocationConfiguration();
-        String rootUrl = globalConfig.getUrl();
-        String buildUrl = "";
-        if (rootUrl == null) {
-            buildUrl = " PLEASE SET JENKINS ROOT URL FROM GLOBAL CONFIGURATION " + build.getUrl();
-        }
-        else {
-            buildUrl = rootUrl + build.getUrl();
+        // JenkinsLocationConfiguration globalConfig = new JenkinsLocationConfiguration();
+        String rootUrl = "";
+        String buildUrl;
+        Jenkins jenkinsInstance = Jenkins.getInstance();
+        if (jenkinsInstance != null ){
+            rootUrl = jenkinsInstance.getRootUrl();
+            if (!rootUrl.isEmpty()){
+                buildUrl = rootUrl + build.getUrl();
+            }else{
+                buildUrl = "Jenkins instance root URL is empty "+ build.getUrl();
```

Comment:
I think the previous message ( 'PLEASE SET JENKINS ROOT URL FROM GLOBAL CONFIGURATION') is more valuable for the user. It says precisely to adjust Jenkins configuration to get the proper build URL.